daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/unparsers/Unparser.scala

```diff
@@ -87,18 +53,28 @@ trait Unparser
 
   def context: RuntimeData
 
-  def unparse(ustate: UState): Unit
+  protected def unparse(ustate: UState): Unit
+
+  final def unparse1(ustate: UState, ignore: AnyRef = null) = ustate.unparse1(this)
 
-  final def unparse1(ustate: UState, context: RuntimeData): Unit = {
+  final def unparse2(ustate: UState): Unit = {
     Assert.invariant(isInitialized)
-    if (ustate.dataProc.isDefined) ustate.dataProc.get.before(ustate, this)
+
     //
     // Since the state is being overwritten (in most case) now,
-    // we must explicitly make a copy so we can compute a delta
+    // we must explicitly make a copy when debugging so we can compute a delta
     // after
     //
-    unparse(ustate)
+    // ?? TODO: Should this be after the split below ??
```

Comment:
Hmmm. This could be why the debugger/trace are displaying things differently. Right now it's copy for debug before a split (if a split happens).